src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
-     */
-    private async commonHandlerWhenValueRejected(
+    private askElicitationQuestion(
         input: ControlInput,
         resultBuilder: ControlResultBuilder,
-    ): Promise<void> {
-        this.state.rejectedValues.push(this.state.value!);
-        const ambiguousPartner = this.getAmbiguousPartner(this.state.value!);
-        if (ambiguousPartner !== undefined && !this.state.rejectedValues.includes(ambiguousPartner)) {
-            const previousValue = this.state.value;
-            this.state.value = ambiguousPartner;
-            const validationResult = await evaluateValidationProp(this.props.validation, this.state, input);
-            if (validationResult === true) {
-                // this is to confirm from users for the suggestedValue
-                this.state.isValidValue = true;
-                this.state.lastInitiative.actName = ConfirmValueAct.name;
-                resultBuilder.addAct(
-                    new SuggestValueAct(this, {
-                        value: this.state.value,
-                        renderedValue:
-                            this.state.value !== undefined
-                                ? this.props.valueRenderer(this.state.value, input)
-                                : '',
-                    }),
-                );
-            } else {
-                this.state.value = previousValue;
-                resultBuilder.addAct(new RequestValueAct(this));
+        elicitationAction: string,
+    ) {
+        this.state.elicitationAction = elicitationAction;
+        this.addInitiativeAct(new RequestValueAct(this), resultBuilder);
+    }
+
+    addInitiativeAct(initiativeAct: InitiativeAct, resultBuilder: ControlResultBuilder) {
+        this.state.lastInitiative.actName = initiativeAct.constructor.name;
+        resultBuilder.addAct(initiativeAct);
+    }
+
+    isConfirmationRequired(input: ControlInput): boolean {
+        if (this.state.value === undefined) {
+            return false;
+        }
+
+        if (this.state.confirmed === true) {
+            return false;
+        }
+        const propValue = this.props.confirmationRequired;
+        return typeof propValue === 'function' ? propValue.call(this, this.state, input) : propValue;
+    }
+
+    private confirmValue(input: ControlInput, resultBuilder: ControlResultBuilder): void {
+        this.state.lastInitiative.actName = ConfirmValueAct.name;
+        resultBuilder.addAct(
+            new ConfirmValueAct(this, {
+                value: this.state.value,
+                renderedValue:
+                    this.state.value !== undefined ? this.props.valueRenderer(this.state.value, input) : '',
+            }),
+        );
+    }
+
+    standardInitiativeHandlers: ControlInitiativeHandler[] = [
+        {
+            name: 'std:elicitValue',
+            canTakeInitiative: this.wantsToElicitValue,
+            takeInitiative: this.elicitValue,
+        },
+        {
+            name: 'std:invalidValue',
```

Comment:
name suggestion:    "RequestReplacementForInvalidValue (built-in)"

Reply:
Updated